src/features/swap/hooks/useIsExpiredSwap.ts

```diff
@@ -0,0 +1,27 @@
+import { useState } from 'react'
+import type { TransactionInfo } from '@safe-global/safe-gateway-typescript-sdk'
+import useInterval from '@/hooks/useInterval'
+import { isSwapTxInfo } from '@/utils/transaction-guards'
+
+const INTERVAL_IN_MS = 10_000
+
+/**
+ * Checks the expiry time of a swap every 10s
+ * and returns true if the swap expired
+ * @param txInfo
+ */
+const useIsExpiredSwap = (txInfo: TransactionInfo) => {
+  const [isExpired, setIsExpired] = useState<boolean>(false)
+
+  const isExpiredSwap = () => {
```

Comment:
The name of this function sounds like it returns a boolen but it actually sets the state.